src/components/FileUpload/hooks.test.js

```diff
@@ -14,80 +12,87 @@ jest.mock('hooks/app', () => ({
+    const setValue = jest.fn();
+
     beforeEach(() => {
-      jest.clearAllMocks();
-      state.mock();
+      setStateSpy = jest.spyOn(React, 'useState').mockImplementation((value) => [value, setValue]);
     });
+
     afterEach(() => {
-      state.resetVals();
+      setStateSpy.mockRestore();
+      jest.clearAllMocks();
     });
 
     it('start with initial state', () => {
       useUploadConfirmModalHooks(props);
-      state.expectInitializedWith(stateKeys.description, '');
-      state.expectInitializedWith(stateKeys.shouldShowError, false);
+      expect(setStateSpy).toHaveBeenCalledWith('');
+      expect(setStateSpy).toHaveBeenCalledWith(false);
     });
 
     it('confirm upload with description', () => {
-      state.mockVal(stateKeys.description, 'description');
+      setStateSpy.mockImplementation(() => ['description', setValue]);
       const out = useUploadConfirmModalHooks(props);
       out.confirmUploadClickHandler();
       expect(props.uploadHandler).toBeCalledWith(props.file, 'description');
     });
 
     it('confirm upload without description', () => {
-      state.mockVal(stateKeys.description, '');
+      setStateSpy.mockImplementation(() => ['', setValue]);
       const out = useUploadConfirmModalHooks(props);
       out.confirmUploadClickHandler();
-      state.expectSetStateCalledWith(stateKeys.shouldShowError, true);
+      expect(setValue).toHaveBeenCalledWith(true);
     });
 
     it('exit handler', () => {
       const out = useUploadConfirmModalHooks(props);
       out.exitHandler();
-      state.expectSetStateCalledWith(stateKeys.shouldShowError, false);
-      state.expectSetStateCalledWith(stateKeys.description, '');
+      expect(setValue).toHaveBeenCalledWith(false);
+      expect(setValue).toHaveBeenCalledWith('');
       expect(props.closeHandler).toBeCalled();
     });
 
     it('on file description change', () => {
       const out = useUploadConfirmModalHooks(props);
       out.onFileDescriptionChange({ target: { value: 'new description' } });
-      state.expectSetStateCalledWith(stateKeys.description, 'new description');
+      expect(setValue).toHaveBeenCalledWith('new description');
     });
   });
 
   describe('useFileUploadHooks', () => {
     const props = {
       onFileUploaded: jest.fn(),
     };
+    let setStateSpy;
+    const setValue = jest.fn();
+
     beforeEach(() => {
-      jest.clearAllMocks();
-      state.mock();
+      setStateSpy = jest.spyOn(React, 'useState').mockImplementation((value) => [value, setValue]);
     });
+
     afterEach(() => {
-      state.resetVals();
+      setStateSpy.mockRestore();
+      jest.clearAllMocks();
     });
 
     it('start with initial state', () => {
       useFileUploadHooks(props);
-      state.expectInitializedWith(stateKeys.uploadArgs, {});
-      state.expectInitializedWith(stateKeys.isModalOpen, false);
+      expect(setStateSpy).toHaveBeenCalledWith({});
+      expect(setStateSpy).toHaveBeenCalledWith(false);
```

Comment:
It'd be great to add comments here similar to the ones added in `FileRenderer/hooks.test.js`.